python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -504,13 +504,22 @@ def queue_monitor_streaming(self) -> None:
     @structured_config_contributor
     def management_api_http(self) -> None:
         """management_api_http set based on management_eapi data-model."""
+        # Flag indicating if we are in ACT Digital Twin mode and if eAPI access in default VRF is enforced
+        act_backdoor_eapi_management: bool = (
+            self.shared_utils.digital_twin and self.inputs.digital_twin.environment == "act" and self.inputs.digital_twin.fabric.act_backdoor_management
+        )
         if self.inputs.management_eapi.enabled:
             self.structured_config.management_api_http.enable_vrfs.append_new(name=self.inputs.mgmt_interface_vrf)
             self.structured_config.management_api_http._update(
                 enable_http=self.inputs.management_eapi.enable_http,
-                enable_https=self.inputs.management_eapi.enable_https,
+                enable_https=self.inputs.management_eapi.enable_https if not act_backdoor_eapi_management else True,
                 default_services=self.inputs.management_eapi.default_services,
             )
+            if act_backdoor_eapi_management and "default" not in self.structured_config.management_api_http.enable_vrfs:
+                self.structured_config.management_api_http.enable_vrfs.append_new(name="default")
+        elif act_backdoor_eapi_management:
+            self.structured_config.management_api_http.enable_https = True
+            self.structured_config.management_api_http.enable_vrfs.append_new(name="default")
```

Comment:
In this code we are checking 3 times the boolean which is hard to follow. We should keep the existing logic and override _after_ if needed:

```suggestion
                enable_https=self.inputs.management_eapi.enable_https,
                default_services=self.inputs.management_eapi.default_services,
            )
            
        # Enforce management access for ACT Digital Twin if required
        if act_preserve_management_access:
            self.structured_config.management_api_http.enable_https = True
            
            if "default" not in self.structured_config.management_api_http.enable_vrfs:
                self.structured_config.management_api_http.enable_vrfs.append_new(name="default")
```

Reply:
Right, logic was actually incorrect. Should be fixed now (moved it to the very bottom of the func)